src/drivers/imx/esai.c

```diff
@@ -410,6 +410,24 @@ static int esai_probe(struct dai *dai)
 	return 0;
 }
 
+static int esai_get_handshake(struct dai *dai, int direction, int stream_id)
+{
+	// TODO
+	return 0;
+}
+
+static int esai_get_fifo(struct dai *dai, int direction, int stream_id)
+{
+	switch (direction) {
+	case DAI_DIR_PLAYBACK:
+	case DAI_DIR_CAPTURE:
+		return dai_fifo(dai, direction); // stream_id is unused
+	default:
+		// If invalid, return NULL
+		return 0;
```

Comment:
return an error and emit trace is best here.

Reply:
Agreed, you'll see the change when I next update this branch (probably in an hour or two).

I strongly suspect the drivers will be ready for review soon (I'm already at the point where the only error I'm currently getting while testing playback is the buffers getting 0 -- I still need to figure out how to compute the buffer sizes in the DMA drivers)